### ISSUE TITLE
Getting Invalid address while fulfilling offers using opensea sdk.

### ISSUE THREAD

**Issue by karanpokar** [ProjectOpenSea/opensea-js#1103]
This is the initialization->

```
const openseaSDK = new OpenSeaSDK(signer.provider, {
      chain: selectedNFT?.chain === 'eth' ? Chain.Mainnet : Chain.Polygon,
      apiKey: 'API_KEY',
    })
    const accepted = await openseaSDK.fulfillOrder({
      order,
      accountAddress: localStorage.getItem('address'),
    })
```
    
    I have added the order response. 

``{
    "order_hash": "0x88ce41889e54ea14ed5d59475dfd73b4a715db4ebf4879f237d1a57b48ea3cd9",
    "chain": "matic",
    "criteria": {
        "collection": {
            "slug": "komet-1"
        },
        "contract": {
            "address": "0x0c66680c8c1a1e2455097f056719e64bcf797089"
        },
        "trait": null
    },
    "protocol_data": {
        "parameters": {
            "offerer": "0x70b030f1f75e36c16695638caa6b45e88cfa757e",
            "offer": [
                {
                    "itemType": 1,
                    "token": "0x7ceB23fD6bC0adD59E62ac25578270cFf1b9f619",
                    "identifierOrCriteria": "0",
                    "startAmount": "600000000000000",
                    "endAmount": "600000000000000"
                }
            ],
            "consideration": [
                {
                    "itemType": 5,
                    "token": "0x0c66680c8c1a1e2455097f056719E64BcF797089",
                    "identifierOrCriteria": "0",
                    "startAmount": "1",
                    "endAmount": "1",
                    "recipient": "0x70b030f1F75E36C16695638cAa6b45e88CFa757e"
                },
                {
                    "itemType": 1,
                    "token": "0x7ceB23fD6bC0adD59E62ac25578270cFf1b9f619",
                    "identifierOrCriteria": "0",
                    "startAmount": "15000000000000",
                    "endAmount": "15000000000000",
                    "recipient": "0x0000a26b00c1F0DF003000390027140000fAa719"
                }
            ],
            "startTime": "1689597065",
            "endTime": "1690201832",
            "orderType": 2,
            "zone": "0x000000e7Ec00e7B300774b00001314B8610022b8",
            "zoneHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
            "salt": "0x360c6ebe0000000000000000000000000000000000000000c0e07fca3df9392a",
            "conduitKey": "0x0000007b02230091a7ed01230072f7006a004d60a8d4e71d599b8104250f0000",
            "totalOriginalConsiderationItems": 2,
            "counter": 0
        },
        "signature": null
    },
    "protocol_address": "0x00000000000000adc04c56bf30ac9d3c0aaf14dc"
}``

This is the error I am getting-->

```
index.js:238 Uncaught (in promise) Error: invalid address (argument="address", value=undefined, code=INVALID_ARGUMENT, version=address/5.7.0)
    at Logger.makeError (index.js:238:1)
    at Logger.throwError (index.js:247:1)
    at Logger.throwArgumentError (index.js:250:1)
    at Module.getAddress (index.js:71:1)
    at isValidProtocol (utils.js:391:1)
    at OpenSeaSDK.<anonymous> (sdk.js:490:1)
    at Generator.next (<anonymous>)
    at sdk.js:27:1
    at new Promise (<anonymous>)
    at __webpack_modules__../node_modules/opensea-js/lib/sdk.js.__awaiter (sdk.js:9:1)
```

I am fetching order using this api->>

`https://api.opensea.io/v2/offers/collection/${slug}`

**Comment by ryanio** [comment]
you should get the order using the sdk `getOrder()` and pass that into `fulfillOrder()`

See the example in README [here](https://github.com/ProjectOpenSea/opensea-js#buying-items):

```typescript
const order = await openseaSDK.api.getOrder({ side: "ask", ... })
const accountAddress = "0x..." // The buyer's wallet address, also the taker
const transactionHash = await openseaSDK.fulfillOrder({ order, accountAddress })
```

**Comment by karanpokar** [comment]
Can you please elaborate on what parameters to pass as I am getting undefined there. It does not fetch collection offers. What should i do?

**Comment by ryanio** [comment]
for collection offers you need to use https://docs.opensea.io/reference/retrieve-collection-offers and https://docs.opensea.io/reference/fulfill-an-offer, i don't think we have support for that in the sdk yet

**Comment by ryanio** [comment]
we should have support for getting and fulfilling collection offers in next release after #1110 is merged, should be within the next couple of days

**Comment by ryanio** [comment]
Getting and fulfilling collection offers is now supported in [v6.1.5](https://github.com/ProjectOpenSea/opensea-js/releases/tag/v6.1.5) is now supported if you'd like to give it a try. Please open a new issue if you are still experiencing errors with your code on how to recreate the issue. Thanks!
